nibabel/cifti2/parse_cifti2.py

```diff
@@ -0,0 +1,587 @@
+# emacs: -*- mode: python-mode; py-indent-offset: 4; indent-tabs-mode: nil -*-
+# vi: set ft=python sts=4 ts=4 sw=4 et:
+### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+#
+#   See COPYING file distributed along with the NiBabel package for the
+#   copyright and license terms.
+#
+### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+from __future__ import division, print_function, absolute_import
+
+from distutils.version import LooseVersion
+
+import numpy as np
+
+from .cifti2 import (Cifti2MetaData, Cifti2Header, Cifti2Label,
+                     Cifti2LabelTable, Cifti2VertexIndices,
+                     Cifti2VoxelIndicesIJK, Cifti2BrainModel, Cifti2Matrix,
+                     Cifti2MatrixIndicesMap, Cifti2NamedMap, Cifti2Parcel,
+                     Cifti2Surface, Cifti2TransformationMatrixVoxelIndicesIJKtoXYZ,
+                     Cifti2Vertices, Cifti2Volume, CIFTI_BRAIN_STRUCTURES,
+                     CIFTI_MODEL_TYPES, _underscore, Cifti2HeaderError)
+from .. import xmlutils as xml
+from ..spatialimages import HeaderDataError
+from ..externals.six import BytesIO
+from ..batteryrunners import Report
+from ..nifti1 import Nifti1Extension, extension_codes, intent_codes
+from ..nifti2 import Nifti2Header, Nifti2Image
+
+
+class Cifti2Extension(Nifti1Extension):
+    code = 32
+
+    def __init__(self, code=None, content=None):
+        Nifti1Extension.__init__(self, code=code or self.code, content=content)
+
+    def _unmangle(self, value):
+        parser = Cifti2Parser()
+        parser.parse(string=value)
+        self._content = parser.header
+        return self._content
+
+    def _mangle(self, value):
+        if not isinstance(value, Cifti2Header):
+            raise ValueError('Can only mangle a Cifti2Header.')
+        return value.to_xml()
+
```

Comment:
One more newline here should clear up the Travis tests.

Appveyor looks like a problem on their end, to me.